server/src/main/java/org/apache/druid/discovery/DruidLeaderClient.java

```diff
@@ -256,15 +257,21 @@ public String findCurrentLeader()
     );
   }
 
-  private String getCurrentKnownLeader(final boolean cached) throws IOException
+  private String getCurrentKnownLeader(final boolean cached)
   {
     final String leader = currentKnownLeader.accumulateAndGet(
         null,
         (current, given) -> current == null || !cached ? pickOneHost() : current
     );
 
     if (leader == null) {
-      throw new IOE("No known server");
+      throw DruidException.forPersona(DruidException.Persona.ADMIN)
+                          .ofCategory(DruidException.Category.NOT_FOUND)
+                          .build(
+                              "No leader server found for NodeRole [%s] and leaderRequestPath [%s].",
```

Comment:
`NodeRole` is a technical internal term. Instead, you can do something like below 
```
"A leader node could not be found for [%s] service. check the logs of [%s] service to validate that service is healthy. ", nodeRole, nodeRole
```